spec/mailers/candidate_mailer_offers_and_rejections_spec.rb

```diff
@@ -0,0 +1,193 @@
+require 'rails_helper'
+
+RSpec.describe CandidateMailer, type: :mailer do
+  include CourseOptionHelpers
+  include ViewHelper
+  include TestHelpers::MailerSetupHelper
+
+  subject(:mailer) { described_class }
+
+  shared_examples 'a mail with subject and content' do |mail, subject, content|
+    let(:email) { described_class.send(mail, @application_choice) }
+
+    it 'sends an email with the correct subject' do
+      expect(email.subject).to include(subject)
+    end
+
+    content.each do |key, expectation|
+      it "sends an email containing the #{key} in the body" do
+        expect(email.body).to include(expectation)
+      end
+    end
+  end
+
+  before do
+    setup_application
+  end
+
+  describe 'send new offer email to candidate' do
+    around do |example|
+      Timecop.freeze(Time.zone.local(2020, 2, 11)) do
+        example.run
+      end
+    end
```

Comment:
Could this be moved outside of the `describe`? We have this in two `describe` blocks, the other is `application choice rejection emails`.